### PR TITLE
Fix issue 552 - Retain Filters between runs

### LINKED ISSUE #552: Retain Filters between runs
It would be useful if the iOS app retained the filters I set between runs of the app. I have no interest in seeing Android videos, so its irritating to have to disable it each time.

### Steps to reproduce

- Open app.
- Tap Filters button
- Change Platforms to iOS  & Swift.
- Leave app for a while
- Re-open app, filters forgotten.

### Expected behavior

I would expect the app to remember what filters I selected until I reset them myself.

### Actual behavior

It forgets the filters I had selected.

### Environment

* iOS Version: 14.4
* raywenderlich.com App Version: 1.0.6
* Device: iPad Pro 10.5"

### PR DESCRIPTION
**Pull request by adesugbaa**
This pull request fixes [Issue 552 - Retain Filters between runs](https://github.com/razeware/emitron-iOS/issues/552)

### CODE REVIEW

File: Emitron/Emitron.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2588,7 +2588,7 @@
 				CURRENT_PROJECT_VERSION = UNKNOWN;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
 				DEVELOPMENT_ASSET_PATHS = "Emitron/Preview\\ Content";
-				DEVELOPMENT_TEAM = KFCNEC27GU;
+				DEVELOPMENT_TEAM = 6HHQWGY2HB;
```
Comment (0xTim) [inline Emitron/Emitron.xcodeproj/project.pbxproj]: We shouldn't be changing the development team

File: Emitron/Emitron/Filters/FilterGroup.swift [hunk 2]
```diff
@@ -40,13 +40,21 @@ struct FilterGroup: Hashable {
   
   /// Update the filters in this group from a set of filters loaded from the data store
   /// - Parameter savedFilters: A Set of filters, often stored in UserSettings
-  func updateFilters(from savedFilters: Set<Filter>) {
+  mutating func updateFilters(from savedFilters: Set<Filter>) {
     // We only care about filters that are from the same group
     let relevantFilters = savedFilters.filter { $0.groupType == type }
     // Let's go through those saved filters and load the relevant data
     relevantFilters.forEach { filter in
       // If we don't have a stored filter that matched the update, ignore it
```
Comment (0xTim) [inline Emitron/Emitron/Filters/FilterGroup.swift]: We should change this comment to reflect the fact we no longer ignore it

**Review by 0xTim — CHANGES_REQUESTED** [review]
Thanks for this! A couple of small changes and then we're good to go

File: Emitron/Emitron.xcodeproj/project.pbxproj (hunk 1 above)
Comment (adesugbaa) [reply Emitron/Emitron.xcodeproj/project.pbxproj]: Reverted the development team

File: Emitron/Emitron/Filters/FilterGroup.swift (hunk 2 above)
Comment (adesugbaa) [reply Emitron/Emitron/Filters/FilterGroup.swift]: I have modified the comment.

**Review by 0xTim — COMMENTED** [review]
Overall this looks great! Would it be possible to add a test to ensure that we keep the filters?

**Comment by adesugbaa** [comment]
I am not sure how this would be tested. I would appreciate any guidance.

**Comment by 0xTim** [comment]
@adesugbaa this line (`libraryRepository = LibraryRepository(repository: repository, contentsService: contentsService, downloadAction: downloadService, syncAction: syncEngine, serviceAdapter: libraryService, filters: filters)`) looks good to check out. Essentially if you trace that through when running tests (or just commenting out that line and any others needed to make it compile) you should hit a failing test to point you in the right direction.

If you can't find anything then shout and we can take a look - it's certainly possible that this doesn't have unit tests yet

**Comment by adesugbaa** [comment]
I am unable to figure out unit tests.

**Comment by VegetarianZombie** [comment]
Is there anything I can do to help move this PR along?

**Comment by 0xTim** [comment]
@VegetarianZombie If you could add some tests that would be awesome!

**Comment by 0xTim** [comment]
@adesugbaa Thanks for the PR! Your commits have been added to #603 and will be part of that PR which has some tests if you'd like to take a look at how that was achieved. 

Sorry it's taken so long to get your stuff merged in! I'll close this in favour of #603

**Pull request closed without merging**